frontend/src/Frontend/Network.hs

```diff
@@ -762,35 +787,32 @@ performNetworkRequestCustom
   -> Event t req
   -> m (Event t (req, [NetworkErrorResult]))
 performNetworkRequestCustom networkL unwrap onReqs =
-    performEventAsync $ ffor onReqs $ \reqs cb ->
-      reportError cb reqs $ do
-        nodeInfos <- getSelectedNetworkInfos networkL
-        void $ liftJSM $ forkJSM $ do
-          r <- traverse (doReqFailover nodeInfos) $ unwrap reqs
-          liftIO $ cb (reqs, r)
+    performEventAsync $ ffor onReqs $ \reqs cb -> do
+      nodeInfos <- getSelectedNetworkInfos networkL
+      void $ liftJSM $ forkJSM $ do
+        r <- traverse (doReqFailover nodeInfos) $ unwrap reqs
+        liftIO $ cb (reqs, r)
   where
-
-    reportError cb reqs m = do
-      er <- runExceptT m
-      case er of
-        Left err -> liftIO $ cb $ (reqs, map (const $ Left err) $ unwrap reqs)
-        Right () -> pure ()
-
+    doReqFailover :: [NodeInfo] -> NetworkRequest -> JSM NetworkErrorResult
     doReqFailover nodeInfos req =
-      go nodeInfos
+      go DList.empty nodeInfos
         where
-          go = \case
+          go :: (DList.DList NetworkError) -> [NodeInfo] -> JSM NetworkErrorResult
+          go errs = \case
             n:ns -> do
               errRes <- doReq (Just n) req
               case errRes of
                 Left err -> do
                   liftIO $ putStrLn $ "Got err: " <> show err
                   if shouldFailOver err
-                               then go ns
-                               else pure errRes
-                Right _ -> pure errRes
-            [] -> doReq Nothing req
-
+                               then go (errs <> DList.singleton err) ns
+                               else pure $ mkResult errs (Left err)
+                Right res -> pure $ mkResult errs (Right res)
+            [] -> mkResult errs <$> doReq Nothing req
+          mkResult errs (Left e) = This (NEL.fromList $ DList.toList (errs <> pure e))
```

Comment:
I'm not convinced of the value of using DList given the length of the list is always going to be fairly short. I'd rather just keep the NonEmpty list and either just pay the append cost or cons everything and reverse it at the end. We can avoid the NEL.fromList that way too.

Reply:
Fixed by d672530